srml/im-online/src/lib.rs

```diff
@@ -507,46 +512,32 @@ impl<T: Trait> session::OneSessionHandler<T::AccountId> for Module<T> {
 	}
 
 	fn on_before_session_ending() {
-		let mut unresponsive = vec![];
-
-		let current_session = <session::Module<T>>::current_index();
-
+		let session_index = <session::Module<T>>::current_index();
 		let keys = Keys::<T>::get();
 		let current_validators = <session::Module<T>>::validators();
 
-		for (auth_idx, validator_id) in current_validators.into_iter().enumerate() {
-			if !Self::is_online_in_current_session_aux(auth_idx as u32, &validator_id) {
-				let full_identification = T::FullIdentificationOf::convert(validator_id.clone())
-					.expect(
```

Comment:
this is a weak and dangerous proof; it should never have been here.

nothing should be able to panic in initialize_block/finalize_block.